pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -202,31 +202,51 @@ func toKubeRuntimeStatus(status *runtimeapi.RuntimeStatus) *kubecontainer.Runtim
 	return &kubecontainer.RuntimeStatus{Conditions: conditions}
 }
 
-// getSeccompProfileFromAnnotations gets seccomp profile from annotations.
-// It gets pod's profile if containerName is empty.
-func (m *kubeGenericRuntimeManager) getSeccompProfileFromAnnotations(annotations map[string]string, containerName string) string {
-	// try the pod profile.
-	profile, profileOK := annotations[v1.SeccompPodAnnotationKey]
+func (m *kubeGenericRuntimeManager) getSeccompProfile(annotations map[string]string, containerName string,
+	podSecContext *v1.PodSecurityContext, containerSecContext *v1.SecurityContext) string {
+	fieldProfile := func(scmp *v1.SeccompProfile) string {
```

Comment:
move fieldProfile and annotationProfile to be standalone functions (can pass in the profileroot path as a param)... it'll make unit tests easier and we don't need to define them as local functions

Reply:
extracted the functions and added a new set of tests around it.